components/Checkbox/src/index.scss

```diff
@@ -1,108 +1,118 @@
-  margin-inline-end: var(--denhaag-checkbox-margin-inline-end);
-  margin-block-start: var(--denhaag-checkbox-margin-block-start);
-  margin-block-end: var(--denhaag-checkbox-margin-block-end);
-  color: var(--denhaag-checkbox-normal-unchecked-color);
-}
-
-.denhaag-checkbox:focus-within::after,
-.denhaag-checkbox--focus::after {
-  content: "";
-  display: block;
+  align-items: center;
+  position: relative;
+}
+
+.denhaag-checkbox__input {
+  z-index: 1;
   position: absolute;
-  top: -2px;
-  bottom: -2px;
-  left: -2px;
-  right: -2px;
-  border: var(--denhaag-focus-border);
-  border-radius: var(--denhaag-border-radius);
+  top: 0;
+  left: 0;
+  width: var(--utrecht-custom-checkbox-size);
+  margin-inline-start: 0;
+  margin-inline-end: 0;
+  margin-block-start: 0;
+  margin-block-end: 0;
+  aspect-ratio: 1;
+  opacity: 0%;
+  cursor: pointer;
 }
 
-.denhaag-checkbox--checked {
-  color: var(--denhaag-checkbox-normal-checked-color);
+.denhaag-checkbox__icon {
+  display: inline-flex;
+  justify-content: center;
+  align-items: center;
+  width: var(--utrecht-custom-checkbox-size);
+  aspect-ratio: 1;
+  background-color: var(--utrecht-custom-checkbox-background-color);
+  border-width: var(--utrecht-custom-checkbox-border-width);
+  border-style: solid;
+  border-color: var(--utrecht-custom-checkbox-border-color);
+  border-radius: var(--utrecht-custom-checkbox-border-radius);
+  box-sizing: border-box;
+  cursor: pointer;
 }
 
-.denhaag-checkbox:hover,
-.denhaag-checkbox--hover {
-  color: var(--denhaag-checkbox-hover-color);
+.denhaag-checkbox__icon .denhaag-icon {
+  display: none;
 }
 
-.denhaag-checkbox--disabled,
-.denhaag-checkbox--disabled:hover,
-.denhaag-checkbox--disabled.denhaag-checbox--hover {
-  color: var(--denhaag-checkbox-disabled-color);
+.denhaag-checkbox__input:disabled {
+  cursor: default;
 }
 
-.denhaag-checkbox--error {
-  color: var(--denhaag-checkbox-error-color);
+.denhaag-checkbox__input:checked + .denhaag-checkbox__icon {
+  background-color: var(--utrecht-custom-checkbox-checked-background-color);
+  border-width: var(--utrecht-custom-checkbox-checked-border-width);
+  border-color: var(--utrecht-custom-checkbox-checked-border-color);
 }
 
-.denhaag-checkbox--error:hover,
-.denhaag-checkbox--error.denhaag-checkbox--hover {
-  color: var(--denhaag-checkbox-error-hover-color);
+.denhaag-checkbox__input:checked + .denhaag-checkbox__icon .denhaag-icon {
+  display: block;
+  font-size: var(--utrecht-custom-checkbox-icon-size);
+  color: var(--utrecht-custom-checkbox-checked-color);
 }
 
-.denhaag-checkbox--error:hover + .denhaag-form-control-label__label,
-.denhaag-checkbox--error.denhaag-checkbox--hover + .denhaag-form-control-label__label {
-  color: var(--denhaag-checkbox-error-hover-color);
+.denhaag-checkbox.denhaag-checkbox--error .denhaag-checkbox__icon {
+  border-width: var(--utrecht-custom-checkbox-invalid-border-width);
+  border-color: var(--utrecht-custom-checkbox-invalid-border-color);
```

Comment:
Same here and for other styles if they already exist as `form-input` tokens. (Mind that we also have a few `denhaag-input` tokens that could be used as fallbacks)